From bf7c9a2b547a5c31497cb3c94e13a83f70dd696f Mon Sep 17 00:00:00 2001
From: Xinyu Chen <xinyu.chen@freescale.com>
Date: Mon, 11 Jun 2012 16:55:34 +0800
Subject: [PATCH 004/281] ENGR00000003 Move display config from
 fsl-proprietary

display configure file is just a file for copying
Move all input device configure files into
device/fsl/common/input

Signed-off-by: Xinyu Chen <xinyu.chen@freescale.com>
---
 common/display/display.conf             |  2 +
 common/display/display_mode.conf        |  5 ++
 common/input/Dell_Dell_USB_Keyboard.idc | 25 ++++++++++
 common/input/Dell_Dell_USB_Keyboard.kl  | 86 +++++++++++++++++++++++++++++++++
 common/input/eGalax_Touch_Screen.idc    |  5 ++
 imx53_smd/eGalax_Touch_Screen.idc       |  5 --
 imx5x/Dell_Dell_USB_Keyboard.idc        | 25 ----------
 imx5x/Dell_Dell_USB_Keyboard.kl         | 86 ---------------------------------
 imx5x/imx53_smd.mk                      |  4 +-
 imx5x/imx5x.mk                          |  4 +-
 imx6/imx6.mk                            | 12 +++--
 imx6/input/Dell_Dell_USB_Keyboard.idc   | 25 ----------
 imx6/input/Dell_Dell_USB_Keyboard.kl    | 86 ---------------------------------
 imx6/input/eGalax_Touch_Screen.idc      |  5 --
 imx6/sabreauto_6q.mk                    |  4 +-
 15 files changed, 136 insertions(+), 243 deletions(-)
 create mode 100644 common/display/display.conf
 create mode 100644 common/display/display_mode.conf
 create mode 100644 common/input/Dell_Dell_USB_Keyboard.idc
 create mode 100644 common/input/Dell_Dell_USB_Keyboard.kl
 create mode 100644 common/input/eGalax_Touch_Screen.idc
 delete mode 100644 imx53_smd/eGalax_Touch_Screen.idc
 delete mode 100644 imx5x/Dell_Dell_USB_Keyboard.idc
 delete mode 100644 imx5x/Dell_Dell_USB_Keyboard.kl
 delete mode 100644 imx6/input/Dell_Dell_USB_Keyboard.idc
 delete mode 100644 imx6/input/Dell_Dell_USB_Keyboard.kl
 delete mode 100644 imx6/input/eGalax_Touch_Screen.idc

diff --git a/common/display/display.conf b/common/display/display.conf
new file mode 100644
index 0000000..bad92b3
--- /dev/null
+++ b/common/display/display.conf
@@ -0,0 +1,2 @@
+mode=1920x1080p-60
+clordepth=32
diff --git a/common/display/display_mode.conf b/common/display/display_mode.conf
new file mode 100644
index 0000000..a2bf5f3
--- /dev/null
+++ b/common/display/display_mode.conf
@@ -0,0 +1,5 @@
+S:1920x1080p-60
+S:1280x720p-60
+D:1024x768p-60
+S:720x576p-50
+S:720x480p-60
diff --git a/common/input/Dell_Dell_USB_Keyboard.idc b/common/input/Dell_Dell_USB_Keyboard.idc
new file mode 100644
index 0000000..2534919
--- /dev/null
+++ b/common/input/Dell_Dell_USB_Keyboard.idc
@@ -0,0 +1,25 @@
+# Copyright (C) 2010 The Android Open Source Project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+touch.deviceType = touchScreen
+touch.orientationAware = 1
+
+keyboard.layout = Dell_Dell_USB_Keyboard
+keyboard.characterMap = Generic
+keyboard.orientationAware = 1
+keyboard.builtIn = 1
+
+cursor.mode = navigation
+cursor.orientationAware = 1
diff --git a/common/input/Dell_Dell_USB_Keyboard.kl b/common/input/Dell_Dell_USB_Keyboard.kl
new file mode 100644
index 0000000..73d7022
--- /dev/null
+++ b/common/input/Dell_Dell_USB_Keyboard.kl
@@ -0,0 +1,86 @@
+key 399   GRAVE
+key 2     1
+key 3     2
+key 4     3
+key 5     4
+key 6     5
+key 7     6
+key 8     7
+key 9     8
+key 10    9
+key 11    0
+key 1     BACK              WAKE_DROPPED
+key 230   SOFT_RIGHT        WAKE
+key 60    SOFT_RIGHT        WAKE
+key 107   ENDCALL           WAKE
+key 62    ENDCALL           WAKE
+key 64    ENDCALL           WAKE
+key 127   MENU              WAKE
+key 59    MENU              WAKE
+key 228   POUND
+key 227   STAR
+key 231   CALL              WAKE
+key 61    CALL              WAKE
+key 67    DPAD_CENTER
+key 108   DPAD_DOWN
+key 103   DPAD_UP
+key 66    HOME              WAKE
+key 105   DPAD_LEFT
+key 106   DPAD_RIGHT
+key 115   VOLUME_UP
+key 114   VOLUME_DOWN
+key 116   POWER             WAKE
+key 211   FOCUS
+key 212   CAMERA
+
+key 16    Q
+key 17    W
+key 18    E
+key 19    R
+key 20    T
+key 21    Y
+key 22    U
+key 23    I
+key 24    O
+key 25    P
+key 26    LEFT_BRACKET
+key 27    RIGHT_BRACKET
+key 43    BACKSLASH
+
+key 30    A
+key 31    S
+key 32    D
+key 33    F
+key 34    G
+key 35    H
+key 36    J
+key 37    K
+key 38    L
+key 39    SEMICOLON
+key 40    APOSTROPHE
+key 14    DEL
+        
+key 44    Z
+key 45    X
+key 46    C
+key 47    V
+key 48    B
+key 49    N
+key 50    M
+key 51    COMMA
+key 52    PERIOD
+key 53    SLASH
+key 28    ENTER
+        
+key 56    ALT_LEFT
+key 42    SHIFT_LEFT
+key 15    TAB
+key 57    SPACE
+key 150   EXPLORER
+key 155   ENVELOPE        
+
+key 12    MINUS
+key 13    EQUALS
+key 215   AT
+
+
diff --git a/common/input/eGalax_Touch_Screen.idc b/common/input/eGalax_Touch_Screen.idc
new file mode 100644
index 0000000..e33a4d7
--- /dev/null
+++ b/common/input/eGalax_Touch_Screen.idc
@@ -0,0 +1,5 @@
+touch.deviceType = touchScreen
+touch.orientationAware = 1
+
+touch.size.calibration = none
+touch.orientation.calibration = none
diff --git a/imx53_smd/eGalax_Touch_Screen.idc b/imx53_smd/eGalax_Touch_Screen.idc
deleted file mode 100644
index 7d58868..0000000
--- a/imx53_smd/eGalax_Touch_Screen.idc
+++ /dev/null
@@ -1,5 +0,0 @@
-touch.deviceType = touchScreen
-touch.orientationAware = 1
-
-cursor.mode = navigation
-cursor.orientationAware = 1
diff --git a/imx5x/Dell_Dell_USB_Keyboard.idc b/imx5x/Dell_Dell_USB_Keyboard.idc
deleted file mode 100644
index 2534919..0000000
--- a/imx5x/Dell_Dell_USB_Keyboard.idc
+++ /dev/null
@@ -1,25 +0,0 @@
-# Copyright (C) 2010 The Android Open Source Project
-#
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
-#
-#      http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-
-
-touch.deviceType = touchScreen
-touch.orientationAware = 1
-
-keyboard.layout = Dell_Dell_USB_Keyboard
-keyboard.characterMap = Generic
-keyboard.orientationAware = 1
-keyboard.builtIn = 1
-
-cursor.mode = navigation
-cursor.orientationAware = 1
diff --git a/imx5x/Dell_Dell_USB_Keyboard.kl b/imx5x/Dell_Dell_USB_Keyboard.kl
deleted file mode 100644
index 73d7022..0000000
--- a/imx5x/Dell_Dell_USB_Keyboard.kl
+++ /dev/null
@@ -1,86 +0,0 @@
-key 399   GRAVE
-key 2     1
-key 3     2
-key 4     3
-key 5     4
-key 6     5
-key 7     6
-key 8     7
-key 9     8
-key 10    9
-key 11    0
-key 1     BACK              WAKE_DROPPED
-key 230   SOFT_RIGHT        WAKE
-key 60    SOFT_RIGHT        WAKE
-key 107   ENDCALL           WAKE
-key 62    ENDCALL           WAKE
-key 64    ENDCALL           WAKE
-key 127   MENU              WAKE
-key 59    MENU              WAKE
-key 228   POUND
-key 227   STAR
-key 231   CALL              WAKE
-key 61    CALL              WAKE
-key 67    DPAD_CENTER
-key 108   DPAD_DOWN
-key 103   DPAD_UP
-key 66    HOME              WAKE
-key 105   DPAD_LEFT
-key 106   DPAD_RIGHT
-key 115   VOLUME_UP
-key 114   VOLUME_DOWN
-key 116   POWER             WAKE
-key 211   FOCUS
-key 212   CAMERA
-
-key 16    Q
-key 17    W
-key 18    E
-key 19    R
-key 20    T
-key 21    Y
-key 22    U
-key 23    I
-key 24    O
-key 25    P
-key 26    LEFT_BRACKET
-key 27    RIGHT_BRACKET
-key 43    BACKSLASH
-
-key 30    A
-key 31    S
-key 32    D
-key 33    F
-key 34    G
-key 35    H
-key 36    J
-key 37    K
-key 38    L
-key 39    SEMICOLON
-key 40    APOSTROPHE
-key 14    DEL
-        
-key 44    Z
-key 45    X
-key 46    C
-key 47    V
-key 48    B
-key 49    N
-key 50    M
-key 51    COMMA
-key 52    PERIOD
-key 53    SLASH
-key 28    ENTER
-        
-key 56    ALT_LEFT
-key 42    SHIFT_LEFT
-key 15    TAB
-key 57    SPACE
-key 150   EXPLORER
-key 155   ENVELOPE        
-
-key 12    MINUS
-key 13    EQUALS
-key 215   AT
-
-
diff --git a/imx5x/imx53_smd.mk b/imx5x/imx53_smd.mk
index 56f3744..6d52866 100644
--- a/imx5x/imx53_smd.mk
+++ b/imx5x/imx53_smd.mk
@@ -13,8 +13,8 @@ PRODUCT_COPY_FILES += \
 	device/fsl/imx53_smd/init.usb.rc:root/init.freescale.usb.rc \
         device/fsl/imx53_smd/vold.fstab:system/etc/vold.fstab \
 	device/fsl/imx53_smd/gpsreset.sh:system/etc/gpsreset.sh \
-	device/fsl/imx53_smd/eGalax_Touch_Screen.idc:system/usr/idc/eGalax_Touch_Screen.idc \
-	device/fsl/imx53_smd/eGalax_Touch_Screen.idc:system/usr/idc/HannStar_P1003_Touchscreen.idc 
+	device/fsl/common/input/eGalax_Touch_Screen.idc:system/usr/idc/eGalax_Touch_Screen.idc \
+	device/fsl/common/input/eGalax_Touch_Screen.idc:system/usr/idc/HannStar_P1003_Touchscreen.idc 
 
 PRODUCT_COPY_FILES += \
     frameworks/base/data/etc/tablet_core_hardware.xml:system/etc/permissions/tablet_core_hardware.xml \
diff --git a/imx5x/imx5x.mk b/imx5x/imx5x.mk
index d0b5f91..80284ac 100644
--- a/imx5x/imx5x.mk
+++ b/imx5x/imx5x.mk
@@ -232,8 +232,8 @@ PRODUCT_PACKAGES += libubi ubinize ubiformat ubiattach ubidetach ubiupdatevol ub
 PRODUCT_AAPT_CONFIG := normal mdpi
 
 PRODUCT_COPY_FILES +=	\
-	device/fsl/imx5x/Dell_Dell_USB_Keyboard.kl:system/usr/keylayout/Dell_Dell_USB_Keyboard.kl \
-	device/fsl/imx5x/Dell_Dell_USB_Keyboard.idc:system/usr/idc/Dell_Dell_USB_Keyboard.idc \
+	device/fsl/common/input/Dell_Dell_USB_Keyboard.kl:system/usr/keylayout/Dell_Dell_USB_Keyboard.kl \
+	device/fsl/common/input/Dell_Dell_USB_Keyboard.idc:system/usr/idc/Dell_Dell_USB_Keyboard.idc \
 	device/fsl/imx5x/init.rc:root/init.rc \
 	device/fsl/imx5x/initlogo.rle:root/initlogo.rle \
 	external/linux-firmware-imx/firmware/vpu/vpu_fw_imx53.bin:system/lib/firmware/vpu/vpu_fw_imx53.bin
diff --git a/imx6/imx6.mk b/imx6/imx6.mk
index 6d110e0..f8f5024 100644
--- a/imx6/imx6.mk
+++ b/imx6/imx6.mk
@@ -235,16 +235,18 @@ PRODUCT_PACKAGES += \
 PRODUCT_LOCALES += hdpi mdpi
 
 PRODUCT_COPY_FILES +=	\
-	device/fsl/imx6/input/Dell_Dell_USB_Keyboard.kl:system/usr/keylayout/Dell_Dell_USB_Keyboard.kl \
-	device/fsl/imx6/input/Dell_Dell_USB_Keyboard.idc:system/usr/idc/Dell_Dell_USB_Keyboard.idc \
-	device/fsl/imx6/input/eGalax_Touch_Screen.idc:system/usr/idc/eGalax_Touch_Screen.idc \
-	device/fsl/imx6/input/eGalax_Touch_Screen.idc:system/usr/idc/HannStar_P1003_Touchscreen.idc \
+	device/fsl/common/input/Dell_Dell_USB_Keyboard.kl:system/usr/keylayout/Dell_Dell_USB_Keyboard.kl \
+	device/fsl/common/input/Dell_Dell_USB_Keyboard.idc:system/usr/idc/Dell_Dell_USB_Keyboard.idc \
+	device/fsl/common/input/eGalax_Touch_Screen.idc:system/usr/idc/eGalax_Touch_Screen.idc \
+	device/fsl/common/input/eGalax_Touch_Screen.idc:system/usr/idc/HannStar_P1003_Touchscreen.idc \
 	device/fsl/imx6/etc/init.rc:root/init.rc \
 	device/fsl/imx6/etc/apns-conf.xml:system/etc/apns-conf.xml \
 	device/fsl/imx6/etc/init.usb.rc:root/init.freescale.usb.rc \
 	device/fsl/imx6/etc/ueventd.freescale.rc:root/ueventd.freescale.rc \
 	device/fsl/imx6/etc/init.gprs-pppd:system/etc/init.gprs-pppd \
-	device/fsl/imx6/etc/ota.conf:system/etc/ota.conf
+	device/fsl/imx6/etc/ota.conf:system/etc/ota.conf \
+	device/fsl/common/display/display.conf:system/etc/display.conf \
+	device/fsl/common/display/display_mode.conf:system/etc/display_mode.conf
 
 PRODUCT_COPY_FILES +=	\
 	external/linux-firmware-imx/firmware/vpu/vpu_fw_imx6d.bin:system/lib/firmware/vpu/vpu_fw_imx6d.bin 	\
diff --git a/imx6/input/Dell_Dell_USB_Keyboard.idc b/imx6/input/Dell_Dell_USB_Keyboard.idc
deleted file mode 100644
index 2534919..0000000
--- a/imx6/input/Dell_Dell_USB_Keyboard.idc
+++ /dev/null
@@ -1,25 +0,0 @@
-# Copyright (C) 2010 The Android Open Source Project
-#
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
-#
-#      http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-
-
-touch.deviceType = touchScreen
-touch.orientationAware = 1
-
-keyboard.layout = Dell_Dell_USB_Keyboard
-keyboard.characterMap = Generic
-keyboard.orientationAware = 1
-keyboard.builtIn = 1
-
-cursor.mode = navigation
-cursor.orientationAware = 1
diff --git a/imx6/input/Dell_Dell_USB_Keyboard.kl b/imx6/input/Dell_Dell_USB_Keyboard.kl
deleted file mode 100644
index 73d7022..0000000
--- a/imx6/input/Dell_Dell_USB_Keyboard.kl
+++ /dev/null
@@ -1,86 +0,0 @@
-key 399   GRAVE
-key 2     1
-key 3     2
-key 4     3
-key 5     4
-key 6     5
-key 7     6
-key 8     7
-key 9     8
-key 10    9
-key 11    0
-key 1     BACK              WAKE_DROPPED
-key 230   SOFT_RIGHT        WAKE
-key 60    SOFT_RIGHT        WAKE
-key 107   ENDCALL           WAKE
-key 62    ENDCALL           WAKE
-key 64    ENDCALL           WAKE
-key 127   MENU              WAKE
-key 59    MENU              WAKE
-key 228   POUND
-key 227   STAR
-key 231   CALL              WAKE
-key 61    CALL              WAKE
-key 67    DPAD_CENTER
-key 108   DPAD_DOWN
-key 103   DPAD_UP
-key 66    HOME              WAKE
-key 105   DPAD_LEFT
-key 106   DPAD_RIGHT
-key 115   VOLUME_UP
-key 114   VOLUME_DOWN
-key 116   POWER             WAKE
-key 211   FOCUS
-key 212   CAMERA
-
-key 16    Q
-key 17    W
-key 18    E
-key 19    R
-key 20    T
-key 21    Y
-key 22    U
-key 23    I
-key 24    O
-key 25    P
-key 26    LEFT_BRACKET
-key 27    RIGHT_BRACKET
-key 43    BACKSLASH
-
-key 30    A
-key 31    S
-key 32    D
-key 33    F
-key 34    G
-key 35    H
-key 36    J
-key 37    K
-key 38    L
-key 39    SEMICOLON
-key 40    APOSTROPHE
-key 14    DEL
-        
-key 44    Z
-key 45    X
-key 46    C
-key 47    V
-key 48    B
-key 49    N
-key 50    M
-key 51    COMMA
-key 52    PERIOD
-key 53    SLASH
-key 28    ENTER
-        
-key 56    ALT_LEFT
-key 42    SHIFT_LEFT
-key 15    TAB
-key 57    SPACE
-key 150   EXPLORER
-key 155   ENVELOPE        
-
-key 12    MINUS
-key 13    EQUALS
-key 215   AT
-
-
diff --git a/imx6/input/eGalax_Touch_Screen.idc b/imx6/input/eGalax_Touch_Screen.idc
deleted file mode 100644
index e33a4d7..0000000
--- a/imx6/input/eGalax_Touch_Screen.idc
+++ /dev/null
@@ -1,5 +0,0 @@
-touch.deviceType = touchScreen
-touch.orientationAware = 1
-
-touch.size.calibration = none
-touch.orientation.calibration = none
diff --git a/imx6/sabreauto_6q.mk b/imx6/sabreauto_6q.mk
index efc8742..3f76e24 100644
--- a/imx6/sabreauto_6q.mk
+++ b/imx6/sabreauto_6q.mk
@@ -12,8 +12,8 @@ PRODUCT_COPY_FILES += \
 	device/fsl/sabreauto_6q/init.rc:root/init.freescale.rc \
 	device/fsl/sabreauto_6q/init.usb.rc:root/init.freescale.usb.rc \
         device/fsl/sabreauto_6q/vold.fstab:system/etc/vold.fstab \
-	device/fsl/imx6/input/eGalax_Touch_Screen.idc:system/usr/idc/eGalax_Touch_Screen.idc \
-	device/fsl/imx6/input/eGalax_Touch_Screen.idc:system/usr/idc/HannStar_P1003_Touchscreen.idc
+	device/fsl/common/input/eGalax_Touch_Screen.idc:system/usr/idc/eGalax_Touch_Screen.idc \
+	device/fsl/common/input/eGalax_Touch_Screen.idc:system/usr/idc/HannStar_P1003_Touchscreen.idc
 
 # GPU files
 
-- 
1.8.0

